### PR TITLE
Fix HTML structure and add failure test workflow

### PR DESCRIPTION
**Pull request by imrineeman**
## PR Type:
Enhancement

___
## PR Description:
- Corrected the HTML structure in index.html by fixing tags.
- Added a new GitHub Actions workflow to test CI failure upon push events.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

- `index.html`: - Removed a redundant opening tag from an <h2> element.
- Corrected an <h3> element by removing extra content.
- Added a new <h4> element with content "hehihi".
- `.github/workflows/main.yml`: - Introduced a new workflow named "Test exiting on failure".
- Workflow triggers on push events.
- Defines a build job that runs on the latest versions of Ubuntu, Windows, and macOS.
- Includes a step to checkout the repository.
- Adds a step designed to fail with "exit 1" to test CI failure.
- Includes a conditional step that should not execute, echoing a message if it erroneously does.
</details>

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: Test exiting on failure
+
+on: [push]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+    runs-on: ${{ matrix.os }}
```
Comment (itsemri) [inline .github/workflows/main.yml:10]: What is this??

Comment (itsemri) [reply .github/workflows/main.yml:10]: Also this!!

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Test exiting on failure
+
+on: [push]
+
+jobs:
```
Comment (itsemri) [inline .github/workflows/main.yml:5]: Hey!

Comment (itsemri) [reply .github/workflows/main.yml:5]: This is bad..

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: Test exiting on failure
+
+on: [push]
```
Comment (imrineeman) [inline .github/workflows/main.yml:3]: 123

Comment (imrineeman) [inline .github/workflows/main.yml:3]: Can we try a more concise name?

**Pull request closed without merging**